server/src/termdb.matrix.js

```diff
@@ -178,7 +186,14 @@ async function getSampleData(q) {
 				const values = data.gene2sample2value.get(tw.term.gene)
 				const value = Number(values[sampleId])
 				let key = value
-				if (tw.q?.mode == 'discrete') key = getBinLabel(tw.q, lst, value)
+				if (ranges.length) {
+					const filterBin = getBin(ranges, value)
+					if (filterBin == -1) continue
```

Comment:
See here, if the value does not fall in the filter range the sample it is not added